cmd/guaccollect/cmd/deps_dev.go

```diff
@@ -117,6 +131,7 @@ you have access to read and write to the respective blob store.`,
 		}
 
 		initializeNATsandCollector(ctx, opts.pubsubAddr, opts.blobAddr, opts.publishToQueue)
+		shutdown(ctx)
```

Comment:
Should this be via a `defer` as soon as we create it? Right now (at least for this function) it doesn't seem to make a difference but to future proof when we add some code path that returns early

Reply:
Yeah, in a number of cases (this one) we are not using special timing on shutdown and a defer would be better.

Other cases we should call shutdown when we get the `C-c` signal but before we call the context cancel function. I'll look through and see which should be change and which are more sensitive.

I think the main thing for the shutdown is to flush any metrics that are recorded but not sent (the send happens in batches). This will be more applicable in some of the `guacone` commands that use the instrumented clients but do no poll.